lametro/views.py

```diff
@@ -271,8 +283,24 @@ class LAMetroEventsView(EventsView):
     template_name = 'lametro/events.html'
 
     def get_context_data(self, **kwargs):
+        '''
+        The EventsView returns all Events – often, filtered and sorted.
+
+        We prefetch EventMedia (i.e., 'media_urls') in the Events querysets: 
+        this makes for a more efficient page load. 
+        We also order the 'media_urls' by label, ensuring that links to SAP audio
+        come after links to English audio. 'mediaqueryset' facilitates 
+        the ordering of prefetched 'media_urls'.
+        '''
         context = super(LAMetroEventsView, self).get_context_data(**kwargs)
 
+        mediaqueryset = LAMetroEventMedia.objects.annotate(
```

Comment:
how would you feel about moving this prefetch up into the existing event manager [as an extra method](https://docs.djangoproject.com/en/2.1/topics/db/managers/#adding-extra-manager-methods), e.g. `with_media`? that way we have an easy way to access, without repeating code. 

with that change, you can then get the events with media prefetched, like so:

```python
LAMetroevent.objects.with_media().filter(...)   # etc.
```

as far as i can tell, we still get the gains? (i've been looking at the django debug toolbar – are you monitoring a different way? this was my q from semaphor, btw!)

Reply:
sounds excellent! (yeah, I was not pleased about the level of DRY-ness here.) 

And yes, I am using django debug toolbar!